pkg/build/builder/common.go

```diff
@@ -82,3 +92,124 @@ func updateBuildRevision(c client.BuildInterface, build *api.Build, sourceInfo *
 		glog.Warningf("An error occurred saving build revision: %v", err)
 	}
 }
+
+// randomBuildTag generates a random tag used for building images in such a way
+// that the built image can be referred to unambiguously even in the face of
+// concurrent builds with the same name in the same namespace.
```

Comment:
but we should never have a container name collision.   By the same reasoning that if the build tags are not actually unique, then " the purpose of being able to identify with certainty what image has been produced by the current build." is not actually being met.